### PR TITLE
Revert "Reorganizes and splits shuttle code"

### PR DESCRIPTION
**Pull request by Mothblocks**
Reverts tgstation/tgstation#88228

Suspected of causing mass flaky CI failures with:

```
[03:14:33] Runtime in code/modules/shuttle/mobile_port/shuttle_rotate_callbacks.dm,94: list index out of bounds
  proc name: shuttleRotate (/obj/machinery/atmospherics/shuttleRotate)
  src: the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air/layer4)
  src.loc: the plating (94,124,7) (/turf/open/floor/plating)
  call stack:
  the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air/layer4): shuttleRotate(270, null)
  the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air/layer4): afterShuttleMove(hyperspace (214,22,12) (/turf/open/space/transit), /list (/list), 8, 1, 2, 270)
  the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air/layer4): afterShuttleMove(hyperspace (214,22,12) (/turf/open/space/transit), /list (/list), 8, 1, 2, 270)
  the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air/layer4): afterShuttleMove(hyperspace (214,22,12) (/turf/open/space/transit), /list (/list), 8, 1, 2, 270)
  White Ship (/obj/docking_port/mobile): cleanup runway(Deep Space (/obj/docking_port/stationary/picked/whiteship), /list (/list), /list (/list), /list (/list), /list (/list), 270, 2, Space (/area/space))
  White Ship (/obj/docking_port/mobile): initiate docking(Deep Space (/obj/docking_port/stationary/picked/whiteship), 2, 0)
  Shuttle (/datum/controller/subsystem/shuttle): action load(NT Science Vessel (/datum/map_template/shuttle/whiteship/pubby), Deep Space (/obj/docking_port/stationary/picked/whiteship), 0)
  Deep Space (/obj/docking_port/stationary/picked/whiteship): load roundstart()
  Shuttle (/datum/controller/subsystem/shuttle): setup shuttles(/list (/list))
  Shuttle (/datum/controller/subsystem/shuttle): Initialize()
  Master (/datum/controller/master): init subsystem(Shuttle (/datum/controller/subsystem/shuttle))
  Master (/datum/controller/master): Initialize(10, 0, 1)

```

### CODE REVIEW

File: code/__HELPERS/shuttle.dm [hunk 1]
```diff
@@ -1,52 +0,0 @@
-/// Helper proc that tests to ensure all whiteship templates can spawn at their docking port, and logs their sizes
-/// This should be a unit test, but too much of our other code breaks during shuttle movement, so not yet, not yet.
```
Comment (Mothblocks) [inline code/__HELPERS/shuttle.dm:2]: :thinking:

Comment (Mothblocks) [reply code/__HELPERS/shuttle.dm:2]: image

**Comment by timothymtorres** [comment]
So I just created a PR that loads all the space ruins for our CI/CD:
- #88403

[It passed all of it's tests. ](https://github.com/tgstation/tgstation/actions/runs/12211006230/job/34067956876) ✅ So this means the cause might be something else...

**Comment by Profakos** [comment]
The runtime error is two years old, #71092. 
However, there is definitely a runtime that started popping up after that PR was merged, seemingly caused by one cryo cell re-registering area_power_change a second time after the shuttle moves.

```
[17:19:49] Runtime in code/datums/signals.dm,43: area_power_change overridden. Use override = TRUE to suppress this warning.
Target: Abandoned Ship Medbay (/area/shuttle/abandoned/medbay) Existing Proc: power_change New Proc: power_change
  proc name:  stack trace (/proc/_stack_trace)
  src: null
  call stack:
   stack trace("area_power_change overridden. ...", "code/datums/signals.dm", 43)
  the omni machinery (/obj/machinery/atmospherics/components/unary): RegisterSignal(Abandoned Ship Medbay (/area/shuttle/abandoned/medbay), "area_power_change", "power_change", 0)
  the omni machinery (/obj/machinery/atmospherics/components/unary): on enter area(the omni machinery (/obj/machinery/atmospherics/components/unary), Abandoned Ship Medbay (/area/shuttle/abandoned/medbay))
  the omni machinery (/obj/machinery/atmospherics/components/unary):  SendSignal("enter_area", /list (/list))
  Abandoned Ship Medbay (/area/shuttle/abandoned/medbay): Entered(the omni machinery (/obj/machinery/atmospherics/components/unary), Space (/area/space))
  the omni machinery (/obj/machinery/atmospherics/components/unary): doMove(the floor (195,89,12) (/turf/open/floor/iron))
  the omni machinery (/obj/machinery/atmospherics/components/unary): forceMove(the floor (195,89,12) (/turf/open/floor/iron))
  /datum/gas_machine_connector (/datum/gas_machine_connector): moved connected machine(the cryo cell (/obj/machinery/cryo_cell), the floor (169,27,14) (/turf/open/floor/iron), 5, 1, null, 0)
  the cryo cell (/obj/machinery/cryo_cell):  SendSignal("movable_moved", /list (/list))
  the cryo cell (/obj/machinery/cryo_cell): Moved(the floor (169,27,14) (/turf/open/floor/iron), 5, 1, null, 0)
  the cryo cell (/obj/machinery/cryo_cell): abstract move(the floor (195,89,12) (/turf/open/floor/iron))
  the cryo cell (/obj/machinery/cryo_cell): onShuttleMove(the floor (195,89,12) (/turf/open/floor/iron), the floor (169,27,14) (/turf/open/floor/iron), /list (/list), 8, null, Hospital Ship (/obj/docking_port/mobile))
  Hospital Ship (/obj/docking_port/mobile): takeoff(/list (/list), /list (/list), /list (/list), 0, 8, null, Space (/area/space))
  Hospital Ship (/obj/docking_port/mobile): initiate docking(Deep Space (/obj/docking_port/stationary/picked/whiteship), 8, 0)
  Shuttle (/datum/controller/subsystem/shuttle): action load(Hospital Ship (/datum/map_template/shuttle/whiteship/box), Deep Space (/obj/docking_port/stationary/picked/whiteship), 0)
  Deep Space (/obj/docking_port/stationary/picked/whiteship): load roundstart()
  Shuttle (/datum/controller/subsystem/shuttle): setup shuttles(/list (/list))
  Shuttle (/datum/controller/subsystem/shuttle): Initialize()
  Master (/datum/controller/master): init subsystem(Shuttle (/datum/controller/subsystem/shuttle))
  Master (/datum/controller/master): Initialize(10, 0, 1)```

**Comment by distributivgesetz** [comment]
> However, there is definitely a runtime that started popping up after that PR was merged, seemingly caused by one cryo cell re-registering area_power_change a second time after the shuttle moves.

Here's the runtime happening in CI
https://github.com/tgstation/tgstation/actions/runs/12174950776/job/33958357444
Here's the proof that the runtime happens without #88228 applied (checks belong to this commit head)
https://github.com/tgstation/tgstation/tree/fd0feb38b84e7f328b924eca26f3a55e31014b7d/code/modules/shuttle

**Comment by distributivgesetz** [comment]
I'm gonna say this in here again too
I did not touch shuttle code itself at all, there were zero additions outside of comments, #88228 should not be the cause and doesn't need to be reverted

**Comment by SmArtKar** [comment]
The issue stems from the Pubby whiteship

image

which has an air tank, which is a ``QUATERNARY``device

```js
/obj/machinery/atmospherics/components/tank
	icon = 'icons/obj/pipes_n_cables/stationary_canisters.dmi'
	icon_state = "smooth"

	name = "pressure tank"
	desc = "A large vessel containing pressurized gas."

	device_type = QUATERNARY
```

It is marked as such because you can open and close nodes on any of its sides

```js
    for(var/i in 1 to length(nodes))
        var/obj/machinery/atmospherics/components/node = nodes[i]
        if(!node)
            continue
        if(src in node.nodes)
            node.disconnect(src)
        nodes[i] = null
        if(parents[i])
            nullify_pipenet(parents[i])
```

As seen in this screenshot by @tontyGH (Thanks for cooperating with me on the investigation), ``Find`` in ``shuttleRotate`` seems to fail on this particular structure, produces a 0 which then is passed as an index without any validation. 

image

Need to dig deeper as to what exactly here is causing the failure, but its related to the this specific quirk of the tank as per https://github.com/tgstation/tgstation/issues/71092#issuecomment-1806830475

**Comment by SmArtKar** [comment]
My bet on what caused the failure would be #87910, fits both the timeline and the affected area. It just made the ruin actually spawn during CI, which broke it.

**Comment by Mothblocks** [comment]
@distributivgesetz FWIW reordering the files *can* cause slight adjustments in proc call orders etc so in something as tangled up as shuttle code it's not unlikely that just reordering would cause behavior changes--I checked the diffs while making the PR and that was my theory

That being said I think smartkar is right

**Pull request closed without merging**